### DISCUSSION TITLE
Get Text from Window

### DISCUSSION THREAD

**Discussion by UUoocl** [WarmUpTill/SceneSwitcher#715] [Ideas]
Thanks for reading my previous idea.  Could Advance Scene Switcher read the text content of a window?  Using a window's content as a condition or action would be useful for OBS automation.  

The Windows tool [Power Automate Desktop](https://powerautomate.microsoft.com/en-us/robotic-process-automation/) has an action named "Get details of window".  The text of the window is then stored in a variable.  Here are examples of the text returned from a PowerPoint slide show 
![image](https://user-images.githubusercontent.com/99063397/230724521-d7df8267-7ff3-40e9-b46c-2dbbaf8221f6.png)

and the Advanced Scene Switcher  
![image](https://user-images.githubusercontent.com/99063397/230724258-717b56b0-7a4a-4dc6-ad6a-a4916afd90c9.png)

Would it possible for Advance Scene switcher to have a similar option to use a window's text content?

**Comment by WarmUpTill** [comment]
Thanks for the suggestion - I will try to look into it! :)

While not quite what you are looking for, note that in the recent release the option to run "[OCR](https://github.com/WarmUpTill/SceneSwitcher/wiki/Video-condition#contains-text)" over a given video source was added.
Depending on what exactly you are trying to achieve this _might_ already help you.

**Reply by UUoocl** [reply]
Thanks for the feedback.  I'm trying to read text that isn't displayed on the screen. In the PowerPoint example, an object could be given a specific name to trigger a scene change.  The object isn't actually displayed on the screen, but is used to trigger an action.

I think the text that Power Automate Desktop is returning may also be used by accessibility screen readers.

**Comment by Destroy666x** [comment]
You  could use AHK script and run that from CLI.

**Comment by WarmUpTill** [comment]
Just for completeness, as I never got back to this thread:

On Windows the "Window" condition has the option to check for text contained in a given window.
It uses the UIAutomation API of Windows.
It does not seem to be quite as powerful as the "Power Automate Desktop" feature mentioned above, however.

**Reply by UUoocl** [reply]
Thanks for reviewing my request.  I tried using the "OCR" feature and it is an excellent solution.  Thanks again for the Scene Switcher.
